+Base/pref.m

```diff
@@ -227,34 +285,45 @@ function set_ui_value(obj,val)
                 % the validate method; no validation necessary here
                 addParameter(p,'default',obj.default);
             end
+            
+            % Go through all public properties (removing Hidden and Abstract ones)
             mc = metaclass(obj);
             mps = mc.PropertyList;
-            for i = 1:nprops % Need to bypass get methods using the metaprop
-                mp = mps(strcmp(props{i},{mps.Name}));
-                assert(mp.HasDefault && iscell(mp.DefaultValue) && length(mp.DefaultValue)==2,...
-                    'Default value of "%s" should be cell array: {default, validation_function}',props{i});
-                addParameter(p,props{i},mp.DefaultValue{1},mp.DefaultValue{2});
+            nprops = length(mps);
+            for ii = 1:nprops % Need to bypass get methods using the metaprop
+                mp = mps(ii);
+                if ~mp.Hidden && ~mp.Abstract
+                    assert(mp.HasDefault && iscell(mp.DefaultValue) && length(mp.DefaultValue)==2,...
+                        'Default value of "%s" should be cell array: {default, validation_function}',mp.Name);
+                    addParameter(p,mp.Name,mp.DefaultValue{1},mp.DefaultValue{2});
+                end
             end
             parse(p,varargin{:});
             if default_in_parser
                 default = p.Results.default;
             end
+            
             % Assign props
-            for i = 1:nprops
-            	obj.(props{i}) = p.Results.(props{i});
+            for ii = 1:nprops
+                mp = mps(ii);
+                if ~mp.Hidden && ~mp.Abstract
+                    obj.(mp.Name) = p.Results.(mp.Name);
+                end
             end
```

Comment:
Changed to go entirely through metaclass rather than properties. Much faster now. Also changed to ignore validation from Hidden functions. Not sure if I've thought of all the repercussions of this, but it works fine.